src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,86 @@
+package duke.command;
+
+import duke.exception.InvalidArgumentException;
+
+import duke.parser.Action;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+import duke.ui.Messages;
+import duke.ui.Ui;
+
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+
+public class AddCommand extends Command {
+
+    private Action action;
+    private String input;
+
+    public AddCommand(Action action, String input) {
+        this.action = action;
+        this.input = input;
+    }
+
+    /**
+     * Adds a task to the Tasklist.
+     *
+     * @param tasks TaskList that command is executed on.
+     *
+     * @param ui User interface that interacts with the user.
+     *
+     * @param storage Storage that saves and loads tasks after Command is executed.
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+            switch (action) {
+            case TODO:
+                try {
+                    String description = Parser.parseDescription(this.input);
+                    tasks.add(new Todo(description));
+                    ui.printTaskAdded(tasks);
+                    storage.save(tasks);
+                } catch (InvalidArgumentException e) {
+                    ui.showError(e.getMessage());
+                } catch (IOException e) {
+                    ui.showError(Messages.SAVE_ERROR);
+                } finally {
+                    break;
+                }
```

Comment:
I think it would be better if you put this whole chunk as another function, so you would not have a very long switch case. (Since there might be more cases in the future)

Reply:
Great idea 👍  will do it when I have the time!